consensus_encoding/src/compact_size.rs

```diff
@@ -88,16 +88,12 @@ pub fn encode(value: impl ToU64) -> ArrayVec<u8, MAX_ENCODING_SIZE> {
 /// * Panics in release mode if the `slice` does not contain a valid minimal compact size encoding.
 /// * Panics in debug mode if the encoding is not minimal (referred to as "non-canonical" in Core).
```

Comment:
These docs are stale, we must have had `debug_assert!` in there at some stage. The panics are not conditional on debug/relase mode anymore.